src/ClusterProviders/Proto.Cluster.Consul/ConsulProvider.cs

```diff
@@ -189,7 +189,8 @@ private void UpdateTtl()
                 let memberIdKey = $"{_clusterName}/{v.Service.Address}:{v.Service.Port}"
                 let memberId = GetMemberId(memberIdKey)
                 where memberId != null
-                let passing = Equals(v.Checks[1].Status, HealthStatus.Passing)
+                //Check #1 is the check we have declared, check 0 is the Consul Serf Status
+                let passing = v.Checks.Length > 1 && Equals(v.Checks[1].Status, HealthStatus.Passing)

```

Comment:
This fixes the status exception that was seen in some cases. simply a race condition where the custom health check had not yet been registered/become visible to external parties